### PR TITLE
feat: quickstart argilla-server

### LINKED ISSUE #2152: Improve argilla docker template image
Simplify all work setting up the argilla docker image for the argilla huggingface space 

https://huggingface.co/docs/hub/spaces-sdks-docker-argilla

https://huggingface.co/spaces/argilla/template-space-docker/blob/main/Dockerfile

### PR DESCRIPTION
**Pull request by rhoitjadhav**
## Description

Quickstart argilla server using docker

Closes #2152

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

**How Has This Been Tested**
Tested locally

**Checklist**

- [ ] I have merged the original branch into my forked branch
- [ ] I added relevant documentation
- [x] follows the style guidelines of this project
- [x] I did a self-review of my code
- [x] I added comments to my code
- [ ] I made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: scripts/start_quickstart_argilla.sh
Comment (dvsrepo) [inline scripts/start_quickstart_argilla.sh]: For the quickstart, I would remove the Hugging Face references/naming. Also as the workspace will be reflected in the UI as `admin` let's call this user `team` for example so it's more friendly as a workspace name.

File: scripts/start_quickstart_argilla.sh
Comment (frascuchon) [inline scripts/start_quickstart_argilla.sh]: We need 2 different credentials configuration:

1. The admin username, should be private to quickstart deployers
2. The argilla username that can be shared to login with.

We should have something like:

`ADMIN_PASSWORD`
`ADMIN_API_KEY`
`ARGILLA_PASSWORD`
`ARGILLA_API_KEY`

File: quickstart.Dockerfile
Comment (frascuchon) [inline quickstart.Dockerfile]: This will install the latest published release. We should have something similar to the `release.Dockerfile` image. Copy the generated wheels packages and install them locally.

Comment (dvsrepo) [reply quickstart.Dockerfile]: Yes, this image should reuse/resemble as much as possible the other image.

File: quickstart.Dockerfile
Comment (frascuchon) [inline quickstart.Dockerfile]: Did you try with an official python image as a base image? Maybe make sense if it works, since could simplify image maintenance.

File: quickstart.Dockerfile
Comment (frascuchon) [inline quickstart.Dockerfile]: It could be also interesting expose the default elasticsearch port (9200)

File: quickstart.Dockerfile
Comment (frascuchon) [inline quickstart.Dockerfile]: Will this command install also the elasticsearch deps like java virtual machine and all the stuff?

File: scripts/start_quickstart_argilla.sh
Comment (dvsrepo) [reply scripts/start_quickstart_argilla.sh]: @rhoitjadhav keep in mind that these ENV vars should all have defaults, and these defaults should be aligned with the defaults we have when launching Argilla locally (user credentials and API key). This is highly important for the user experience of these quickstart users both for the local and HF deployment: we don't require them to set anything or generate passwords if they just want to quickly launch an Argilla instance for experimentation. We provide these ENVs to give more flexibility to more advanced quickstart users that want to better restrict access to datasets.

For the API_KEYS:
`ADMIN_API_KEY`: this is the one that should be used if you want to keep a secret API key for uploading data. This should be then mapped with the API_KEY of HF Spaces deployments.
`ARGILLA_API_KEY` should be hidden from public users of HF Spaces (i.e., not visible in the code for launching the Space) otherwise random users would be able to add datasets to the Space.

For PASSWORDS:
The role of the ARGILLA user is to be able to login into the UI and explore/label data. The ARGILLA user must have access to the workspace where datasets with `ADMIN_API_KEY` are uploaded.
The credentials should be the default ones we have currently for local installations. If quickstart users want to restrict access to random users they can set the `ARGILLA_PASSWORD` as a secret. 

For `ADMIN_PASSWORD` I don't have any comments, and for the ADMIN_USERNAME either. I think the most important thing is to keep it a no-brainer to launch a default quickstart instance.

File: quickstart.Dockerfile
Comment (rhoitjadhav) [reply quickstart.Dockerfile]: Yes it does. This step is suggested in their [docs](https://www.elastic.co/guide/en/elasticsearch/reference/7.17/deb.html#deb-repo) also.

File: quickstart.Dockerfile
Comment (rhoitjadhav) [reply quickstart.Dockerfile]: Sure I'll add that

File: quickstart.Dockerfile
Comment (rhoitjadhav) [reply quickstart.Dockerfile]: I think I should, actually for elasticsearch 8.5 its giving error (unrecognized service) when we try to start the elasticsearch so I'm thinking to change the image to Python 3.9 or may be I'll use ES 8.5 image and test the same.

File: quickstart.Dockerfile
Comment (rhoitjadhav) [reply quickstart.Dockerfile]: Sure I'll use that

File: quickstart.Dockerfile
Comment (frascuchon) [reply quickstart.Dockerfile]: great!

File: quickstart.Dockerfile
Comment (frascuchon) [reply quickstart.Dockerfile]: Okay. Let me know if you have some troubles

**Comment by rhoitjadhav** [comment]
@frascuchon made changes as per your suggestions

File: quickstart.Dockerfile
Comment (frascuchon) [inline quickstart.Dockerfile]: Duplicated statement

File: scripts/start_quickstart_argilla.sh
Comment (frascuchon) [inline scripts/start_quickstart_argilla.sh]: This message is shown when launching the container. Could we hide it?

**Review by frascuchon — CHANGES_REQUESTED** [review]
I've run it locally and it's working.

The only things to remark on (and we need to review):
- After running the container
```bash
docker run -p 6900:6900 -p 9200:9200 \
-e ADMIN_PASSWORD=admin -e ARGILLA_PASSWORD=1234 \
-e ADMIN_API_KEY=admin -e ARGILLA_API_KEY=argilla \
--name argilla-quickstart \
-d clay-quickstart
```
The container cannot be stopped with a Ctrl+C

- Stopping the container from the Docker dashboard and then rerunning it makes it crash (and no info is raised in logs)

A one-shot execution can be enough, but we should solve problems for a better user experience.

File: scripts/start_quickstart_argilla.sh
Comment (frascuchon) [inline scripts/start_quickstart_argilla.sh]: Could we disable this as part of the docker image build, since we need setup it up just once?

**Review by dvsrepo — CHANGES_REQUESTED** [review]

File: scripts/start_quickstart_argilla.sh
Comment (rhoitjadhav) [reply scripts/start_quickstart_argilla.sh]: Yes its hidden

File: scripts/start_quickstart_argilla.sh
Comment (rhoitjadhav) [reply scripts/start_quickstart_argilla.sh]: Its done

File: quickstart.Dockerfile
Comment (rhoitjadhav) [reply quickstart.Dockerfile]: Done

**Comment by frascuchon** [comment]
This PR closes issue #2152. Please, include this info in PR description.

**Review by frascuchon — CHANGES_REQUESTED** [review]
It looks good to me.

Just a comment to apply. We will use the next patch release to create the first version of this new image. This will simplify integrations with HF spaces. 

So please, change the target branch from `develop` to `main` (be sure to keep only changes applied in this branch when you change the branch, you should apply a commit cherry-pick for that)

**Comment by frascuchon** [comment]
The branch contains changes from develop that won't be included in this PR. Please, put here only changes affecting the quickstart docker image ( you can use cherry-picking to select desired commits instead of a merge from develop)

cc @rhoitjadhav

**Comment by rhoitjadhav** [comment]
yes will do that.

File: scripts/start_quickstart_argilla.sh
Comment (dvsrepo) [inline scripts/start_quickstart_argilla.sh]: @rhoitjadhav as discussed. Let's change this variable to `LOAD_DATASETS` and use the following options:

1. `full`: it loads all current datasets from load_data.py
2. `single`: it loads just the `sst-sentiment-explainability` (that is until line 53 of `load_data.py`
3. `none`: it doesn't load any dataset

**Review by dvsrepo — CHANGES_REQUESTED** [review]

File: scripts/start_quickstart_argilla.sh
Comment (rhoitjadhav) [reply scripts/start_quickstart_argilla.sh]: Pushed the changes and deployed on HF

Comment (dvsrepo) [reply scripts/start_quickstart_argilla.sh]: 💯 !

**Comment by dvsrepo** [comment]
This ready from my side 🚀

**Comment by rhoitjadhav** [comment]
@frascuchon Added environment variables section in the quickstart docs file. Please review and let me know if there anything to improve otherwise I think we can merge this and I believe after merging it will generate `latest` tag docker image. [`argilla/argilla-quickstart:latest`]

**Review by frascuchon — APPROVED** [review]

**Review by dvsrepo — APPROVED** [review]

**Pull request merged**
